src/main/java/org/apache/flink/scheduler/benchmark/SchedulerBenchmarkUtils.java

```diff
@@ -0,0 +1,245 @@
+			ExecutionMode executionMode) throws IOException {
+
+		final JobGraph jobGraph = new JobGraph(jobVertices.toArray(new JobVertex[0]));
+
+		jobGraph.setScheduleMode(scheduleMode);
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		executionConfig.setExecutionMode(executionMode);
+		jobGraph.setExecutionConfig(executionConfig);
+
+		return jobGraph;
+	}
+
+	public static ExecutionGraph createExecutionGraph(
+			JobGraph jobGraph,
+			SlotProvider slotProvider) throws IOException {
+
+		final JobInformation jobInformation = new DummyJobInformation(
+				jobGraph.getJobID(),
+				jobGraph.getName());
+
+		final ClassLoader classLoader = ExecutionGraph.class.getClassLoader();
+		return new ExecutionGraph(
+				jobInformation,
+				TestingUtils.defaultExecutor(),
+				TestingUtils.defaultExecutor(),
+				AkkaUtils.getDefaultTimeout(),
+				new NoRestartStrategy(),
+				JobManagerOptions.MAX_ATTEMPTS_HISTORY_SIZE.defaultValue(),
+				new RestartAllStrategy.Factory(),
+				slotProvider,
+				classLoader,
+				VoidBlobWriter.getInstance(),
+				Time.seconds(10L),
+				new RegionPartitionReleaseStrategy.Factory(),
+				NettyShuffleMaster.INSTANCE,
+				NoOpJobMasterPartitionTracker.INSTANCE,
+				jobGraph.getScheduleMode(),
+				NoOpExecutionDeploymentListener.INSTANCE,
+				(execution, newState) -> {
+				},
+				System.currentTimeMillis());
+	}
+
+	public static void waitForListFulfilled(
+			Collection<?> list,
+			int length,
+			long maxWaitMillis) throws TimeoutException {
+
+		final Deadline deadline = Deadline.fromNow(Duration.ofMillis(maxWaitMillis));
+		final Predicate<Collection<?>> predicate = (Collection<?> l) -> l.size() == length;
+		boolean predicateResult;
+
+		do {
+			predicateResult = predicate.test(list);
+
+			if (!predicateResult) {
+				try {
+					Thread.sleep(2L);
+				} catch (InterruptedException ignored) {
+					Thread.currentThread().interrupt();
+				}
+			}
+		} while (!predicateResult && deadline.hasTimeLeft());
+
+		if (!predicateResult) {
+			throw new TimeoutException(String.format(
+					"List no fulfilled in time, expected %d, actual %d.",
+					length,
+					list.size()));
+		}
+	}
+
+	public static void verifyListSize(
+			Collection<?> list,
+			int length) {
+		if (list.size() < length) {
+			throw new RuntimeException(String.format(
+					"Size of the list mismatch, expected %d, actual %d.",
+					length,
+					list.size()));
+		}
+	}
+
+	public static void deployTasks(
+			ExecutionGraph executionGraph,
+			JobVertexID jobVertexID,
+			TestingLogicalSlotBuilder slotBuilder,
+			boolean sendScheduleOrUpdateConsumersMessage) throws Exception {
+
+		for (ExecutionVertex vertex : executionGraph.getJobVertex(jobVertexID).getTaskVertices()) {
+			LogicalSlot slot = slotBuilder.createTestingLogicalSlot();
+			vertex.getCurrentExecutionAttempt()
+					.registerProducedPartitions(
+							slot.getTaskManagerLocation(),
+							sendScheduleOrUpdateConsumersMessage).get();
+			vertex.deployToSlot(slot);
```

Comment:
This method is no longer used in production and is possibly to be removed soon.
Maybe we can add a method `deployTask(Execution, LogicalSlot)` here which invokes `Execution#tryAssignResource` and `Execution#deployToSlot`.

Reply:
> This method is no longer used in production and is possibly to be removed soon.
> Maybe we can add a method `deployTask(Execution, LogicalSlot)` here which invokes `Execution#tryAssignResource` and `Execution#deployToSlot`.

Which one is deprecated, `ExecutionVertex#deployToSlot` or `Execution#registerProducedPartitions`? 

`ExecutionVertex#deployToSlot` is used by `DefaultExecutionVertexOperations#deploy`. 

`Execution#registerProducedPartition` is used by `DefaultScheduler#assignResourceOrHandleError`.

Reply:
`ExecutionVertex#deployToSlot()` is not used in production. What `DefaultExecutionVertexOperations#deploy()` invokes is `ExecutionVertex#deploy()`.

Reply:
> `ExecutionVertex#deployToSlot()` is not used in production. What `DefaultExecutionVertexOperations#deploy()` invokes is `ExecutionVertex#deploy()`.

Yes, I find that `deployToSlot` is only used in test cases. I've fixed it.